### DISCUSSION TITLE
Issues about compiling MohrCoulomb’s model as external soil model

### DISCUSSION THREAD

**Discussion by 19Dario88** [Anura3D/Anura3D_OpenSource#227] [Q&A]
Hi everybody, 
I’m studying how to compile and build an External Soil Model, using Visual Studio 2022 (v.17.9.6), integrated by OneAPI toolkits (base and advanced), as suggested by the ANURA3D's Manual. With this software's configuration, when I compile the code, everything is ok. 
However, when I try to build a DLL starting from the  UMAT and ESM ".f" files of the Mohr Coulomb's model (in their original version, placed in “.\Anura3D_OpenSource-main\src\Soilmodels”), Visual Studio reports several errors, and the DLL is not created. Here attached, the log reporting the errors. 
Is it a compatibility problem between the files and the version of Visual Studio+API toolkits, or there are errors in the “.f” files stored in the ANURA3D's folder? Thank you in advance for any hints or answers! 
Regards,
Dario
[BuildLog.txt](https://github.com/user-attachments/files/15914850/BuildLog.txt)
![Visual Studio screenshot](https://github.com/Anura3D/Anura3D_OpenSource/assets/168671577/93c70b9b-3039-49f1-b0fa-cc0dc7e2c0d0)

**Comment by 19Dario88** [comment]
After configuring again Visual Studio according to the ANURA3D's manual, only 2 errors remain..How can I solve them?
![Visual Studio screenshot](https://github.com/Anura3D/Anura3D_OpenSource/assets/168671577/6d62fb61-435d-453c-b293-4d0b9b7e54ea)

**Comment by ReySonic** [comment]
I recommend you use the external material models from the gid program, at least I find it much more intuitive than from vs
